ansible/roles/esdc-common/files/zoneinit

```diff
@@ -0,0 +1,94 @@
+#!/bin/bash
+
+# Copyright 2013, Joyent. Inc. All rights reserved.
+
+PATH=/opt/local/bin:/opt/local/sbin:/usr/bin:/usr/sbin:/bin:/sbin
+export PATH
+
+set -o errexit
+set -o xtrace
+
+. /lib/svc/share/smf_include.sh
+smf_is_globalzone && exit ${SMF_EXIT_OK}
+
+# Set basedir and default config file
+ZONEINIT_INCLUDES=/var/zoneinit/includes
+ZONECONFIG=/root/zoneconfig
+
+# This script should only continue if
+#
+#  1) executed as a SMF start method; and
+#  2) /var/svc/provisioning exist.
+#
+# Otherwise, pretend success but do nothing
+# except updating authorized_keys of root user
+#
+
+AUTHORIZED_KEYS="$(mdata-get root_authorized_keys || echo '')"
```

Comment:
Please add a comment with a simple explanation why we are doing this.

(A link to `https://github.com/erigones/esdc-factory/issues/43` should be enough)